custom_md_loader/classes/MetadataApexApiLoader.cls

```diff
@@ -0,0 +1,230 @@
+			String typeDevName = mappingInfo.getCustomMetadadataTypeName()
+						.subString(0, mappingInfo.getCustomMetadadataTypeName().indexOf(AppConstants.MDT_SUFFIX));
+	        List<Metadata.CustomMetadata> records = new List<Metadata.CustomMetadata>();            
+			for(sObject csRecord : mappingInfo.getRecordList()) {
+			
+	            Metadata.CustomMetadata customMetadataRecord = new Metadata.CustomMetadata();
+	            customMetadataRecord.values = new List<Metadata.CustomMetadataValue>();
+	            
+	            if(csRecord.get(AppConstants.CS_NAME_ATTRIBURE) != null) {
+	            	String strippedLabel = (String)csRecord.get(AppConstants.CS_NAME_ATTRIBURE);
+	            	String tempVal = strippedLabel.substring(0, 1);
+	                
+	                if(tempVal.isNumeric()) {
+	                	strippedLabel = 'X' + strippedLabel;
+	                }
+	            	strippedLabel = strippedLabel.replaceAll('\\W+', '_').replaceAll('__+', '_').replaceAll('\\A[^a-zA-Z]+', '').replaceAll('_$', '');
+	                System.debug('strippedLabel ->' + strippedLabel);
+	                
+	                // default fullName to type_dev_name.label
+					customMetadataRecord.fullName = typeDevName + '.'+ strippedLabel;
+					customMetadataRecord.label = (String)csRecord.get(AppConstants.CS_NAME_ATTRIBURE);
+	            }
+	            for(String fieldName : mappingInfo.getCSToMDT_fieldMapping().keySet()) {
+		            Schema.DescribeFieldResult descCSFieldResult = descFieldResultMap.get(fieldName.toLowerCase());
+	            
+	            	if(mappingInfo.getCSToMDT_fieldMapping().get(fieldName).endsWith('__c')){
+	        			Metadata.CustomMetadataValue cmv = new Metadata.CustomMetadataValue();
+	                	cmv.field = mappingInfo.getCSToMDT_fieldMapping().get(fieldName);
+	                	if(descCSFieldResult.getType().name() == 'DATETIME') {
+	                	
+	                		if(csRecord.get(fieldName) != null) {
+								Datetime dt = DateTime.valueOf(csRecord.get(fieldName));
+								String formattedDateTime = dt.format('yyyy-MM-dd\'T\'HH:mm:ss.SSSZ'); 
+								cmv.value = csRecord.get(formattedDateTime);
+							}
+							else {
+								cmv.value = null;
+							}
+							
+						}
+						else{
+		            		cmv.value = csRecord.get(fieldName);
+	            		}
+	                	
+	                	customMetadataRecord.values.add(cmv);
+	            	}
+	        	}
+	            records.add(customMetadataRecord);
+	        }
+	            
+			
+		    callback.setMdDeployStatus(mdDeployStatus);
+		    
+			Metadata.DeployContainer deployContainer = new Metadata.DeployContainer();
+			for(Metadata.CustomMetadata record : records) {
+	        	deployContainer.addMetadata(record);
+	        }
+			
+			// Enqueue custom metadata deployment
+	    	Id jobId = Metadata.Operations.enqueueDeployment(deployContainer, callback);
+	    	jobId1 = jobId;
+	    	

```

Comment:
Maybe at this point after the enqueue, you could add a message to the page telling the user where in setup they could go to see the status/results.